packages/shared/src/common/retry.ts

```diff
@@ -94,6 +94,13 @@ export async function executeWithRetries<T>(
           return result;
         }
         log(`[${context}] Retrying... (attempt: ${attempt})`);
+        const timeToWait = getTimeToWait({
+          initialDelay,
+          backoffMultiplier,
+          attempt,
+          maxDelay,
+        });
+        await sleep(timeToWait);
```

Comment:
This was missing here! We usually don't use this flow, when we retry even when there's no error, that's why we didn't come across this problem before.